pkg/upgrade/upgrade.go

```diff
@@ -83,22 +85,50 @@ func RunUpgrade(OSD *osd.OSD) error {
 
 // TriggerUpgrade uses a helper to perform an upgrade.
 func TriggerUpgrade(h *helper.H) (*configv1.ClusterVersion, error) {
+	var cVersion *configv1.ClusterVersion
+	var err error
 	// setup Config client
 	cfgClient := h.Cfg()
 
 	// get current Version
 	getOpts := metav1.GetOptions{}
-	cVersion, err := cfgClient.ConfigV1().ClusterVersions().Get(ClusterVersionName, getOpts)
+	cVersion, err = cfgClient.ConfigV1().ClusterVersions().Get(ClusterVersionName, getOpts)
 	if err != nil {
 		return cVersion, fmt.Errorf("couldn't get current ClusterVersion '%s': %v", ClusterVersionName, err)
 	}
 
 	// set requested upgrade targets
-	cVersion.Spec.DesiredUpdate = &configv1.Update{
-		Version: strings.Replace(h.Upgrade.ReleaseName, "openshift-v", "", -1),
-		Image:   h.Upgrade.Image,
-		Force:   h.Upgrade.Image != "", // Force if we have an image specified
+	if h.Upgrade.Image != "" {
+		cVersion.Spec.DesiredUpdate = &configv1.Update{
+			Version: strings.Replace(h.Upgrade.ReleaseName, "openshift-v", "", -1),
+			Image:   h.Upgrade.Image,
+			Force:   true, // Force if we have an image specified
+		}
+	} else {
+		upgradeVersion := strings.Replace(h.Upgrade.ReleaseName, "openshift-v", "", -1)
+		installVersion := strings.Replace(state.Instance.Cluster.Version, "openshift-v", "", -1)
+
+		upgradeVersionParsed := semver.MustParse(upgradeVersion)
+		installVersionParsed := semver.MustParse(installVersion)
+
+		if upgradeVersionParsed.Minor() > installVersionParsed.Minor() {
```

Comment:
We should add a TODO here to make sure we address major eventually. This will work for OSD 4.x, but will break for 5.x. If we put a TODO we'll at least have a breacrumb to fix the issue when it actually becomes a problem (years down the line).